addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -314,7 +314,7 @@ export class Many2XAutocomplete extends Component {
         };
     }
     async loadOptionsSource(request) {
-        if (this.lastProm) {
+        if (this.lastProm && this.lastProm.abort) {
```

Comment:
Well it had to do with the fact that doing a .then on a Promise makes it lose its .abort() and my override of .search used .then() so i wanted to avoid potential errors with .abort() not being a function but it is true that it doesn't feel like the right way to do it. I let you take a look at my new way of handling this and tell me what you think.